src/main/java/org/apache/sling/scripting/core/impl/SlingBindingsVariablesListJsonServlet.java

```diff
@@ -91,6 +102,24 @@ protected void activate(ComponentContext context) {
     @Override
     protected void doGet(SlingHttpServletRequest request, SlingHttpServletResponse response)
             throws ServletException, IOException {
+        if (webconsoleSecurity == null) {
+            log("Acccess forbidden as the WebConsoleSecurity reference is not set");
+            response.sendError(HttpServletResponse.SC_FORBIDDEN);
+            return;
+        } else if (!(webconsoleSecurity instanceof WebConsoleSecurityProvider2)) {
+            log("Acccess forbidden as the WebConsoleSecurity reference does not implement WebConsoleSecurityProvider2");
+            response.sendError(HttpServletResponse.SC_FORBIDDEN);
+            return;
+        } else if (!((WebConsoleSecurityProvider2)webconsoleSecurity).authenticate(request, response)) {
+            // the request is terminated without any more response sent back to the client.
+            //    The WebConsoleSecurityProvider2 implementation may have sent auth challenge to the client
+            //    in the case of anonymous access.
+            if (!response.isCommitted()) {
```

Comment:
why the check for isCommitted only here and not for the other two cases?

Reply:
I guess I didn't think there was a code path where the response was touched before the servlet was called.  But I suppose it is possible that some filter has done something before calling the servlet, so I'll change it to do the same check for all the cases.